app/src/main/java/ch/epfl/skysync/models/calendar/AvailabilityCalendar.kt

```diff
@@ -3,8 +3,16 @@ package ch.epfl.skysync.models.calendar
 import ch.epfl.skysync.models.UNSET_ID
 import java.time.LocalDate
 
+/** Represents the type of a cell difference between two calendar */
+enum class CalendarDifferenceType {
```

Comment:
You've implemented a type of logging protocol that keeps track of the local changes before persisting it to the DB. However, I ask myself if it would not be less complicated to just store a list of availabilities and updating the lists by simple remplacement of the whole list...

For a list of availabilities = [1,2,3,4]
If we update two elements of the list to [1', 2',3,4] with the current strategy, after assessing what values have changed, we do two requests to update 1 and 2 individually 

But we might as well overwrite the old list with the new list which would no longer need for every entry in the list to have a proper ID and remove the assessment phase.

In order to limit the size of the list we could have some partitioning of the list by range

Reply:
This could be a tempting solution, except that in addition to the scalability problem, it makes querying availability more difficult (think of the admin use case), even more so if we start partitioning the list.
So I think the current solution is simpler and more versatile (also think that it'll only be a few dozens requests at most).
I already looked if there was a way to batch the requests but didn't find anything, I'll look again when I have the time.

Reply:
What kind of scalability problem do you see? Each array could be max. a month of availabilities <= 2*31 entries in the array. 

The availabilities could then be modeled:
month: 10
year: 2023
userid: someID
array = (ok, ok, maybe, ...)

Reply:
Sure, it is something we can keep in mind, but at the moment, unless we notice that the current setup doesn't work. I advise to stick to what we have and move forward